util.py

```diff
@@ -39,39 +43,55 @@ def toAlpha3Code(code):
         return iso639CodesInverse[code] if code in iso639CodesInverse else code
 
 
-def getLocalizedLanguages(locale, dbPath, languages=[]):
+def getLanguageNames(locale, dbPath):
     global langNamesDBConn
+
     if not langNamesDBConn:
         if os.path.exists(dbPath):
             langNameDBConn = sqlite3.connect(dbPath)
-            c = langNameDBConn.cursor()
+            cursor = langNameDBConn.cursor()
         else:
             logging.error('Failed to locate language name DB: %s' % dbPath)
-            return {}
+            return None
+
+    return cursor.execute('SELECT * FROM languageNames WHERE lg=?', (locale, )).fetchall()
+
+
+@gen.coroutine
+def getLocalizedLanguages(locale, dbPath, languages=[]):
+    languageResults = yield ThreadPoolExecutor(1).submit(getLanguageNames, locale, dbPath)
+
+    if not languageResults:
+        logging.error('Failed to locate language name DB: %s' % dbPath)
```

Comment:
Do we really need this error twice?

Reply:
Sorry, forgot to clean up the artifacts.